### PR TITLE
Gateway subnet metadata synchronization 

### PR DESCRIPTION
**Pull request by jshaofuturewei**
It is for the issue https://github.com/CentaurusInfra/fornax/issues/72

This is to add watchers for new created mizar objects. When a new object, such as subnet is added, it will trigger a creation event caught by watcher. Therefore the watcher can create a remote subnet in another cluster through grpc.

It is supposed to come with the grpc implementation. However, the total changed file list is over 150+. Therefore, I have to split the change into 2 pull requests. 

1. Create a soft link 
```bash
ln -s fornax github.com/kubeedge/kubeedge
```
2. Create the following files as requested
```bash
cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/register.go
cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/v1/doc.go
cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/v1/register.go
cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/v1/types.go
```
3. Run the following command to generate subnet client codes 
```bash
cloud/hack/generate-groups.sh all github.com/kubeedge/kubeedge/cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/v1/ github.com/kubeedge/kubeedge/cloud/cmd/inter_cluster_gateway/pkg/apis/ "subnet:v1"
```
4. Verify if the deepcopy file has been created at cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/v1/
5. Verify the rest client codes at github.com/kubeedge/kubeedge/cloud/cmd/inter_cluster_gateway/pkg/apis/subnet/v1/
6. Updated the main.go to add watchers(TODO, since grpc codes are not come together)
7. Verify that a new subnet event is caught.
```bash
DAP server listening at: 127.0.0.1:2345
WARNING: undefined behavior - Go version 1.13.9 is too old for this version of Delve (minimum supported version 1.15)
I0214 21:53:41.348760   19796 reflector.go:207] Starting reflector *v1.Subnet (0s) from pkg/mod/k8s.io/client-go@v0.19.10/tools/cache/reflector.go:156
I0214 21:53:41.348876   19796 reflector.go:243] Listing and watching *v1.Subnet from pkg/mod/k8s.io/client-go@v0.19.10/tools/cache/reflector.go:156
I0214 21:53:41.366175   19796 main.go:340] A new subnet net0 is trying to sync to 
I0214 21:53:41.366908   19796 main.go:340] A new subnet net1 is trying to sync to 
I0214 21:53:41.367192   19796 main.go:340] A new subnet net2 is trying to sync to 
I0214 21:53:41.367467   19796 main.go:340] A new subnet net3 is trying to sync to 
github.com/kubeedge/kubeedge/cloud/cmd/inter_cluster_gateway/pkg/apis/ "subnet:v1"
```

### CODE REVIEW

**Comment by pdgetrf** [comment]
please add a description about the specific function this PR provides in addition to the list of changes. for example, "this PR provides the function to sync vpc/subnet when xxx and yyy happens."

File: scripts/deployment/create_cluster_gateway_configmap.sh [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
-gateway_filename=cluster_gateway.properties
-
-gateway_host_ip="$(cat ${gateway_filename})"
+source cluster_gateway.properties
 
 # Remove the old configmap file
 rm -f cluster_gatewway_configmap.yaml
 
 # Create a new configmap file from portal)host.properties
```
Comment (pdgetrf) [inline scripts/deployment/create_cluster_gateway_configmap.sh:8]: typo portal)host

File: cloud/cmd/inter_cluster_gateway/util/kube_watcher.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	if _, err = watcher.subnetInterface.Create(context.TODO(), &subnet, metav1.CreateOptions{}); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func GetLocalHostIP() net.IP {
+	conn, err := net.Dial("udp", "8.8.8.8:80")
```
Comment (pdgetrf) [inline cloud/cmd/inter_cluster_gateway/util/kube_watcher.go:180]: what's this 8.8.8.8:80 for?

**Review by pdgetrf — APPROVED** [review]
LGTM

File: cloud/cmd/inter_cluster_gateway/util/kube_watcher.go (hunk 2 above)
Comment (jshaofuturewei) [reply cloud/cmd/inter_cluster_gateway/util/kube_watcher.go]: Removed. Test codes

File: scripts/deployment/create_cluster_gateway_configmap.sh (hunk 1 above)
Comment (jshaofuturewei) [reply scripts/deployment/create_cluster_gateway_configmap.sh:8]: Corrected

**Pull request merged**